### PR TITLE
Move deleted row filtering to pure pipe

### PR DESCRIPTION
**Pull request by GuessWhoSamFoo**
Changes row deleted checking to pure pipe

Signed-off-by: GuessWhoSamFoo <foos@vmware.com>

### CODE REVIEW

**Comment by GuessWhoSamFoo** [comment]
The usage of a pure pipe allows caching row inputs for free so `rowClass` won't be repeatedly called when idle.

File: web/src/app/modules/shared/components/presentation/datagrid/datagrid.component.ts [hunk 1]
```diff
@@ -121,9 +121,9 @@ export class DatagridComponent implements OnChanges {
     return true;
   }
 
-  rowClass(row: TableRowWithMetadata) {
-    return row.isDeleted ? ['row-deleted'] : [];
-  }
+  // rowClass(row: TableRowWithMetadata) {
```
Comment (bryanl) [inline web/src/app/modules/shared/components/presentation/datagrid/datagrid.component.ts:124]: This should be deleted

File: web/src/app/modules/shared/components/presentation/datagrid/datagrid.component.ts [hunk 2]
```diff
@@ -148,7 +148,7 @@ export class DatagridComponent implements OnChanges {
   }
 }
 
-interface TableRowWithMetadata {
+export interface TableRowWithMetadata {
```
Comment (bryanl) [inline web/src/app/modules/shared/components/presentation/datagrid/datagrid.component.ts:151]: since it is exported, could you move this to `content.ts`

File: web/src/app/modules/shared/pipes/filterDeleted/filter-deleted.pipe.spec.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { FilterDeletedPipe } from './filter-deleted.pipe';
+
+describe('FilterDeletedPipe', () => {
```
Comment (bryanl) [inline web/src/app/modules/shared/pipes/filterDeleted/filter-deleted.pipe.spec.ts:3]: This is an easy item to test, so add tests for the positive and negative cases.

File: web/src/app/modules/shared/pipes/filterDeleted/filter-deleted.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { TableRowWithMetadata } from '../../components/presentation/datagrid/datagrid.component';
+
+@Pipe({ name: 'filterDeleted', pure: true })
+export class FilterDeletedPipe implements PipeTransform {
```
Comment (bryanl) [inline web/src/app/modules/shared/pipes/filterDeleted/filter-deleted.pipe.ts:5]: Would it be better if this had a better name? I will suggest `FilterDeletedDatagridRowPipe`
